orca-core/src/main/groovy/com/netflix/spinnaker/orca/pipeline/persistence/jedis/JedisExecutionRepository.groovy

```diff
@@ -135,30 +120,49 @@ class JedisExecutionRepository implements ExecutionRepository {
 
   @Override
   boolean isCanceled(String id) {
+    String key = fetchKey(id)
     withJedis { Jedis jedis ->
-      String key
-      if (jedis.exists("pipeline:$id")) {
-        key = "pipeline:$id"
-      } else if (jedis.exists("orchestration:$id")) {
-        key = "orchestration:$id"
-      } else {
-        throw new ExecutionNotFoundException("No execution found with id $id")
-      }
       Boolean.valueOf(jedis.hget(key, "canceled"))
     }
   }
 
   @Override
+  void pause(String id, String user) {
+    String key = fetchKey(id)
+    withJedis { Jedis jedis ->
+      def pausedDetails = new Execution.PausedDetails(
```

Comment:
I'm going to add a check here that only RUNNING pipelines can be PAUSED ... and only PAUSED pipelines can be resumed.